GitUI/UserControls/ToolStripGitStatus.cs

```diff
@@ -156,6 +205,8 @@ private void TryStartWatchingChanges(string workTreePath, string gitDirPath)
                 {
                     _workTreeWatcher.Path = workTreePath;
                     _gitDirWatcher.Path = gitDirPath;
+                    _globalIgnoreFilePath = DetermineGlobalIgnoreFilePath();
+                    _globalIgnoreWatcher.Path = Path.GetDirectoryName(_globalIgnoreFilePath);
```

Comment:
You have to assign the Path only if the value being assigning is a valid directory, otherwise an exception is thrown.
EnableRaisingEvents should take into account if the Path is empty.